tests/test_examples.py

```diff
@@ -72,6 +73,11 @@ def test_loss_and_gradient_so3():
     def test_loss_and_gradient_se3():
         loss_and_gradient_se3.main()
 
+    @staticmethod
+    @geomstats.tests.np_only
```

Comment:
I know this PR is already huge, so for the next one: it would be great if it can work in tf/torch as well, otherwise this will decrease or test coverage :(

Reply:
I am also working on it, but since I knew there would be major changes, I was waiting a bit to see where that PR would go (although only the numpy version of `eye` allows for defining subdiagonals, which is a pain)

Reply:
Makes sense!

For the eye issue, maybe gs.diag or gs.diagonal can help?